plugins/widget/plugin.js

```diff
@@ -2701,6 +2754,22 @@
 				var attrs = element.attributes,
 					widget, widgetElement;
 
+				// Reset initial and trailing space by replacing &nbsp; with white space (#605).
+				if ( 'data-cke-widget-white-space' in attrs ) {
+					var firstTextNode = getFirstTextNode( element ),
+						lastTextNode = getLastTextNode( element );
+
+					// Check whether the value of the text node contains &nbsp; at the beginning and replace it with white space.
+					if ( firstTextNode.parent.attributes[ 'data-cke-white-space-first' ] && firstTextNode.value.match( /^&nbsp;/g ) ) {
```

Comment:
Is there any reason why `firstTextNode.value.match( /^&nbsp;/g )` is checked here? If the element has `data-cke-white-space-first` attribute it already means that space was replaced there, so the second check seems redundant.